airflow/models/dag.py

```diff
@@ -1678,6 +1679,17 @@ def _test_cycle_helper(self, visit_map, task_id):
 
         visit_map[task_id] = DagBag.CYCLE_DONE
 
+    @provide_session
+    def test_pools(self, available_pools=None, session=None):
+        requested_pools = {task.pool for task in self.task_dict.values() if task.pool}
```

Comment:
If possible, we should avoid adding new methods to DAG. This is the main object of the application and it has too much responsibility.  Could you move this method to any other place?

Reply:
Sure, I moved this to a helper function.